wave_lang/kernel/wave/scheduling/scheduler_utils.py

```diff
@@ -139,20 +140,21 @@ def get_local_writes(self, local_loads):
             local_writes.update(cur_writes)
         return list(local_writes)
 
-    def get_lds_gathers(self, local_loads):
-        lds_gathers = set()
+    def get_global_loads_to_lds(self, local_loads):
+        loads = set()
         for local_load in local_loads:
             custom = get_custom(local_load)
             # Get direct users and users from rotated registers.
             memory_users = set([g for g in custom.memory.users])
-            # Filter users for GatherToLDS
-            cur_gathers = [
+            # Filter users for GatherToLDS or TensorLoadToLDS
+            cur_loads = [
                 g
                 for g in memory_users
-                if isinstance(get_custom(g), GatherToLDS) and g.graph == custom.graph
+                if isinstance(get_custom(g), (GatherToLDS, TensorLoadToLDS))
+                and g.graph == custom.graph
             ]
-            lds_gathers.update(cur_gathers)
-        return list(lds_gathers)
+            loads.update(cur_loads)
+        return list(loads)
```

Comment:
I know it was in the old code, but `set` iteration order is not stable, we should probably use `dict[key, None]`